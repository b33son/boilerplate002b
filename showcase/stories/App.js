import React from 'react'
import { storiesOf, action } from '@storybook/react'
import App from '../../src/components/App'

const toggleSidebar = action('toggleSidebar')

storiesOf('App', module)
  .add('initial state', () => {
    const props = {
      id: '1',
      title: 'A Title',
      toggleSidebar
    }

    return (
      <App {...props} />
    )
  })
