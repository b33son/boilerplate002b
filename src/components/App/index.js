import React from 'react'
import { string, func } from 'prop-types'

const App = ({ id, title, toggleSidebar }) => (
  <div>
    <h1>{`${id}: ${title}`}</h1>
    <button onClick={toggleSidebar}>Toggle Sidebar</button>
  </div>
)

App.propTypes = {
  id: string.isRequired,
  title: string,
  toggleSidebar: func
}

export default App
