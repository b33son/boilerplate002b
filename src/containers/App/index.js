import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'
import App from '../../components/App'
import { toggleSidebar } from '../../actions'

export const mapStateToProps = ({ reducer }) => ({
  id: reducer.id,
  title: reducer.pageTitle
})

export const mapDispatchToProps = (dispatch) => bindActionCreators({
  toggleSidebar: toggleSidebar
}, dispatch)

const AppContainer = connect(mapStateToProps, mapDispatchToProps)(App)
export default AppContainer
