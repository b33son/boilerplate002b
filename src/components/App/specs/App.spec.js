import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import App from '../../App'

describe('<App />', () => {
  let wrapper
  let props

  beforeEach(() => {
    props = {
      id: '1',
      title: 'Hello Talkdesk!',
      toggleSidebar: jest.fn()
    }

    wrapper = shallow(<App {...props} />)
  })

  it('renders correctly', () => {
    const tree = renderer.create(<App {...props} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  describe('when the button is clicked', () => {
    it('calls the toggleSidebar handler', () => {
      const button = wrapper.find('button')

      button.simulate('click')

      expect(props.toggleSidebar).toHaveBeenCalled()
    })
  })
})
