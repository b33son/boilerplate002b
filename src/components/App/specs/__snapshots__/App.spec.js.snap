// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders correctly 1`] = `
<div>
  <h1>
    1: Hello Talkdesk!
  </h1>
  <button
    onClick={[Function]}
  >
    Toggle Sidebar
  </button>
</div>
`;
